playground/infrastructure/helper.py

```diff
@@ -134,49 +155,56 @@ def _get_example(filepath: str, filename: str) -> Example:
     return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED)
 
 
-def _match_pattern(filepath: str) -> bool:
-    """Check file to matching
-
-    Check that file has the correct extension and contains the beam-playground tag.
-
-    Args:
-        filepath: path to the file.
-
-    Returns:
-        True if file matched. False if not
+def _validate(tag: dict, categories_path: str) -> bool:
     """
-    extension = filepath.split(os.extsep)[-1]
-    if extension in SUPPORTED_SDK:
-        tag = get_tag(filepath)
-        if tag:
-            _validate(tag)
-            return True
-    return False
+    Validate all tag's fields
 
-
-def _validate(tag: dict):
-    """Validate all tag's fields
-
-    If some of the fields has incorrect format, raise error
+    Validate that tag contains all required fields and all fields have required format.
 
     Args:
         tag: beam tag to validate
+        categories_path: path to the file with all supported categories.
+
+    Returns:
+        In case tag is valid, True
+        In case tag is not valid, False
     """
     if tag.get("name") is None:
-        raise ValueError("tag doesn't contain name field: " + tag.__str__())
+        logging.error("tag doesn't contain name field: " + tag.__str__())
+        return False
     if tag.get("description") is None:
-        raise ValueError("tag doesn't contain description field: " + tag.__str__())
+        logging.error("tag doesn't contain description field: " + tag.__str__())
+        return False
     if tag.get("multifile") is None:
-        raise ValueError("tag doesn't contain multifile field: " + tag.__str__())
+        logging.error("tag doesn't contain multifile field: " + tag.__str__())
+        return False
     multifile = tag.get("multifile")
     if str(multifile).lower() not in ["true", "false"]:
-        raise ValueError("tag's field multifile is incorrect: " + tag.__str__())
+        logging.error("tag's field multifile is incorrect: " + tag.__str__())
+        return False
     if tag.get("categories") is None:
-        raise ValueError("tag doesn't contain categories field: " + tag.__str__())
+        logging.error("tag doesn't contain categories field: " + tag.__str__())
+        return False
+    categories = tag.get("categories")
+    if type(categories) is not list:
+        logging.error("tag's field categories is incorrect: " + tag.__str__())
+        return False
```

Comment:
Let's go through all checks to find all possible troubles with the given tag. I.e. have a flag that goes to `False` if some check fails and if this flag is `False` at the end of all checks -> return it

Reply:
Changed.